src/main/java/command/Command.java

```diff
@@ -0,0 +1,206 @@
+package command;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import exception.DukeException;
+import javafx.scene.layout.VBox;
+import parser.DateTimeParser;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+import ui.DialogBox;
+import ui.Ui;
+
+/**
+ * <h1>Command class</h1>
+ * Filters out the commandType and runs the executes the appropriate
+ * command based on the input commandType.
+ */
+public class Command {
+    private CommandType commandType;
+    private TaskList tasks;
+    private String input;
+    private Ui ui;
+
+    /**
+     * Creates a Command object to be executed.
+     *
+     * @param commandType type of command to be executed.
+     * @param tasks list of Tasks the user currently has.
+     * @param input the String to be parsed.
+     * @param ui the User Interface that prints out the output.
+     */
+    public Command(CommandType commandType, TaskList tasks, String input, Ui ui) {
+        this.commandType = commandType;
+        this.tasks = tasks;
+        this.input = input;
+        this.ui = ui;
+    }
+
+    private TaskList getMatchingTasks(String keyword) {
+        TaskList newTaskList = new TaskList();
+        for (int i = 0; i < tasks.getSize(); i++) {
+            if (tasks.containsKeyword(i, keyword)) {
+                newTaskList.add(tasks.get(i));
+            }
+        }
+        return newTaskList;
+    }
+
+    private void addTask(Task input) {
+        tasks.add(input);
+    }
+
+    private LocalDateTime stringToLocalDateTime(String deadline) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a");
+        LocalDateTime deadlineDateTime = LocalDateTime.parse(deadline, formatter);
+        return deadlineDateTime;
+    }
+
+    private boolean hasTaskIndex(int taskIndex) {
+        return taskIndex - 1 < this.tasks.getSize() && taskIndex - 1 >= 0;
+    }
```

Comment:
Well done! This method can come in handy for different Commands that operates on a specific index in the task list. Implementing it in the parent class is really smart to prevent code duplication!